src/controllers/student.controller.js

```diff
@@ -0,0 +1,108 @@
+import Student from '../models/student.model.js';
+import extend from 'lodash/extend.js';
+import errorHandler from '../helpers/dbErrorHandler.js';
+import Teacher from '../models/teacher.model.js';
+// import Supply from '../models/supply.model.js';
+// import Donation from '../models/donation.model.js';
+
+/**
+ * Controller functions to be mounted on the Student route
+ */
+
+/**
+ * Use to load the supply object into the Express req object BEFORE
+ * propogating to the next function thats specific to the request
+ * that came in
+ */
+ const studentByID = async (req, res, next, id) => {
+    try {
+        let student = await Student.findById(id);
+        if (!student)
+            return res.status(404).json({
+                error: 'Student not found',
+            });
+        req.student = student;
+        next();
+    } catch (err) {
+        return res.status(400).json({
+            error: 'Could not retrieve this student',
+        });
+    }
+};
+
+const create = async (req,res) => {
+    try {
+        const student = await Student.create(req.body);
+        const teacher_id = student.teacher_id;
```

Comment:
Since teacher_id is required in the Student model line 35 will return an error.  Need to add teacher_id to the req.body then create the student.